pkg/apis/core/validation/shoot.go

```diff
@@ -511,6 +511,7 @@ func ValidateNodeCIDRMaskWithMaxPod(maxPod int32, nodeCIDRMaskSize int32, networ
 func ValidateTotalNodeCountWithPodCIDR(shoot *core.Shoot) field.ErrorList {
 	allErrs := field.ErrorList{}
 
+	// TODO: clarify why this is only called for create requests but not for updates
```

Comment:
cc @shafeeqes (introduced with https://github.com/gardener/gardener/pull/5389).
From the PR description:

> The validation is applied only to new shoots, to not break the existing clusters.

Reply:
CC @timebertt 

I guess we can drop the question PR then?